server/src/main/java/io/druid/client/TimelineInventoryView.java

```diff
@@ -0,0 +1,28 @@
+/*
+* Licensed to Metamarkets Group Inc. (Metamarkets) under one
+* or more contributor license agreements. See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership. Metamarkets licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License. You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*/
+
+package io.druid.client;
+
+import io.druid.query.DataSource;
+import io.druid.timeline.TimelineLookup;
+
+public interface TimelineInventoryView extends InventoryView
```

Comment:
From what I can tell, there is only one instance of implementing `TimelineInventoryView` and it is the `CoordinatorServerView`.  Even in `DatasourcesResourceTest` you are mocking `CoordinatorServerView` instead of implementing this interface.  There are two reasons to add an interface

1) You have multiple implementations and want polymorphic dispatch (in tests or in normal code)
2) You expect people to extend the interface and add their own implementations

Given that there is only one implementation of the interface and it doesn't seem to be necessary for the tests, I don't think (1) applies.  I also do not expect (2) to be an issue, this is an internal implementation rather than something I expect someone to extend.

If you agree with my assessment above, then please remove this interface entirely and just depend on the concrete `CoordinatorServerView`.  If not, please let me know what I'm missing.